### ISSUE TITLE
CS0433 conflict on MSBuild referenced assemblies

### ISSUE THREAD

**Issue by erikpowa** [0xd4d/dnSpy#926]
- What version did you use?
dnSpy **v5.0.7**
- Write down all steps needed to reproduce this issue. Also attach any needed files.
1. Get 1 assembly that uses `Microsoft.Build.Utilities` (so basically whatever MSBuild Task assembly)
2. Edit any class that inherits `Microsoft.Build.Utilities.Task`

```
namespace Foo {
	public class Bar : Microsoft.Build.Utilities.Task {
        }
	public override bool Execute(){ return true; }
}
```

3. You can't compile (if the foo.dll not referencing from the GAC where dnySpy do?)

Example:
foo.dll  using `Microsoft.Build.Utilities.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a` from nuget or whatever
dnSpy using `'Microsoft.Build.Utilities.v4.0, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'` from GAC (?)
**=** CS0433

**Comment by 0xd4d** [comment]
I tried to repro it but I can't. My dll file has two files in the same dir:

- Microsoft.Build.Framework.dll
- Microsoft.Build.Utilities.Core.dll

and I can edit the method without an error.

It has a ref to `Microsoft.Build.Utilities.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a`

You can try closing all files, then drag all files that it depends on to Assembly Explorer and see if that helps. If dnSpy fails to find the referenced DLL file in the same folder as the file, it will check the GAC.

**Comment by erikpowa** [comment]
Darn, I moved only the necessary assemblies to an empty folder and it's compiled there, but when I tried compile along with all the references, I still encountered conflict.  So I went futher more and realized it's one of the reference that causing it, check this out:
[classlibrary2.zip](https://github.com/0xd4d/dnSpy/files/2357817/classlibrary2.zip)

You can't edit `ClassLibrary2 ` if `System.Collections.Immutable.dll` represent.

**Comment by 0xd4d** [comment]
Did you try the same trick, close all files, drag all reference assemblies to Assembly Explorer, and then drag the file you want to edit to Assembly Explorer? Eg. the .NET Framework 4.7.2 files are here:

- `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2`
- `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades`

Related: https://github.com/0xd4d/dnSpy/issues/729

**Comment by erikpowa** [comment]
I tried, but the problem is foreg the `System.Collections.Immutable.dll` is not part of the GAC (?) and dnSpy auto pulling it (ofc if it's next to a lib that you want to edit with dnSpy, or just manually get rid of it if it's not being used anyway at runtime).  

So the point of view for the problem is changed. The problem is now maybe something like the referenced `System.Collections.Immutable.dll` pulls netstandard and the netstandard pulls other version GAC library corresponding it's netstandard version (?) (+ how dnSpy resolves the targetframework) which could lead 2 unidentical targetframework or version, therefore conflict. (but I'm **not** sure actually what's going on in the background).

So probably #729 pretty much.

(Not a big deal, since manually addig/removing references fixes it until #729 .)

**Comment by 0xd4d** [comment]
Closing as dupe of #729
